clients/python-client/python_client/kuberay_job_api.py

```diff
@@ -0,0 +1,196 @@
+                    return None
+
+            if resource and "status" in resource and resource["status"]:
+                return resource["status"]
+            else:
+                log.info("rayjob {} status not set yet, waiting...".format(name))
+                time.sleep(delay_between_attempts)
+                timeout -= delay_between_attempts
+
+        log.info("rayjob {} status not set yet, timing out...".format(name))
+        return None
+
+    def wait_until_job_finished(
+        self,
+        name: str,
+        k8s_namespace: str = "default",
+        timeout: int = 60,
+        delay_between_attempts: int = 5,
+    ) -> bool:
+        """Wait until a Ray job reaches a terminal status.
+
+        This method waits for the job to have a jobStatus field with a terminal value
+        (STOPPED, SUCCEEDED, or FAILED).
+
+        Parameters:
+        - name (str): The name of the Ray job custom resource.
+        - k8s_namespace (str, optional): The namespace in which to retrieve the Ray job. Defaults to "default".
+        - timeout (int, optional): The duration in seconds after which we stop trying. Defaults to 60 seconds.
+        - delay_between_attempts (int, optional): The duration in seconds to wait between attempts. Defaults to 5 seconds.
+
+        Returns:
+            bool: True if the rayjob reaches a terminal status, False otherwise.
+        """
+        while timeout > 0:
+            status = self.get_job_status(
+                name, k8s_namespace, timeout, delay_between_attempts
+            )
+
+            if status and "jobStatus" in status:
+                current_status = status["jobStatus"]
+                if current_status in ["", "PENDING"]:
+                    log.info("rayjob {} has not started yet".format(name))
+                elif current_status == "RUNNING":
+                    log.info("rayjob {} is running".format(name))
+                elif current_status in TERMINAL_JOB_STATUSES:
+                    log.info(
+                        "rayjob {} has finished with status {}!".format(
+                            name, current_status
+                        )
+                    )
+                    return True
+                else:
+                    log.info(
+                        "rayjob {} has an unknown status: {}".format(
+                            name, current_status
+                        )
+                    )
+            else:
+                log.info(
+                    "rayjob {} jobStatus field not available yet, waiting...".format(
+                        name
+                    )
+                )
+
+            time.sleep(delay_between_attempts)
+            timeout -= delay_between_attempts
+
+        log.info(
+            "rayjob {} has not reached terminal status before timeout".format(name)
+        )
+        return False
+
+    def delete_job(self, name: str, k8s_namespace: str = "default") -> Any:
+        try:
```

Comment:
for consistency sake - can you add a bit more documentation about this function, same as for `get_job_status` function